### ISSUE TITLE
`lib`: aur-gendb

### ISSUE THREAD

**Issue by AladW** [AladW/aurutils#855]
This is something I've experimented with a while ago: a tool to convert `.SRCINFO` files to a pacman database. This could be used similarly to a `Usage = Search` database, e.g. to search installation candidates, display package information or resolve dependencies with `pacman`.

Practically speaking, a pacman database needs very few fields to be valid. `%NAME%` and `%VERSION` seem to suffice, for `%SIZE%` an arbitrary value can also be set (like 0). With recent `aurweb` developments to publish metadata dumps, optional fields like `%DEPENDS%` can also be populated. 

In short `aur-depends`, `aur-graph` and `aur-search` can be directly replaced by `pacman` (!) using this approach. The same woud likely hold for `aur-vercmp`.

**Comment by Morganamilo** [comment]
Keep I'm mind pacman does not load make/check deps for syncdbs.

**Comment by AladW** [comment]
I guess that's expected. Why then were these fields added in the first place?

**Comment by Morganamilo** [comment]
Consistency I guess. Alpm will load them from the localdb but not for syncdbs. Point of is to not use a ton of extra memory.

**Comment by AladW** [comment]
@Morganamilo seems a database generated this way doesn't work with e.g. `alpm_find_dbs_satisfier`, and `pacsift --repo` and `pactree` find no packages. 

A `desc` entry looks like this:
```
%FILENAME%
aurutils-9.6-3-any.pkg.tar.xz

%NAME%
aurutils

%BASE%
aurutils

%VERSION%
9.6-3

%DESC%
helper tools for the arch user repository

%PACKAGER%
Alad

%BUILDDATE%
1651159328

%DEPENDS%
git
jq
pacutils
curl
expect

%OPTDEPENDS%
bash-completion
zsh
devtools
vifm
ninja

%LICENSE%
custom:ISC

%CSIZE
0

%ISIZE
0

%MD5SUM%
d41d8cd98f00b204e9800998ecf8427e

%SHA256SUM%
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```
Any idea what's missing here?

**Comment by Morganamilo** [comment]
I'm not quite sure but send me the full DB and I'll take a look when at a computer.

**Comment by AladW** [comment]
https://paste.xinu.at/aez/

warning: 660mb unextracted

**Comment by AladW** [comment]
I don't think I'll keep pursuing this:
1. `pacman` performs poorly on a db of that size, and is outperformed by remote queries;
2. `gendb` overlaps in scope with `repo-add` and tools like `repod`;
3. apart from `pactree` (which doesn't work) it adds little beyond demonstration purposes.

If at some point you find out the reason why `pactree` doesn't work, you're welcome to mention it either way. Might come handy in the future.
